grpc/src/main/java/io/stargate/grpc/service/interceptors/metrics/MetricCollectingServerCall.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.grpc.service.interceptors.metrics;
+
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.micrometer.core.instrument.Counter;
+
+/**
+ * A simple forwarding server call that collects metrics.
+ *
+ * @param <Q> The type of message received one or more times from the client.
+ * @param <A> The type of message sent one or more times to the client.
+ * @author Daniel Theuke (daniel.theuke@heuboe.de)
+ */
+class MetricCollectingServerCall<Q, A> extends SimpleForwardingServerCall<Q, A> {
+
+  private final Counter responseCounter;
```

Comment:
Isn't `Meter` a better type here? It will give us `Counter` + rates.

Reply:
This is directly from Micrometer